patches/net/minecraft/client/renderer/RenderStateShard.java.patch

```diff
@@ -0,0 +1,11 @@
+--- a/net/minecraft/client/renderer/RenderStateShard.java
++++ b/net/minecraft/client/renderer/RenderStateShard.java
+@@ -23,7 +_,7 @@
+     public static final float VIEW_SCALE_Z_EPSILON = 0.99975586F;
+     public static final double MAX_ENCHANTMENT_GLINT_SPEED_MILLIS = 8.0;
+     public final String name;
+-    public final Runnable setupState;
++    protected Runnable setupState; // Neo/AT-BUG: 'protected-f' does not work as field is also made public via AT field group
```

Comment:
I don't like this (irrespective of the AT/JST bug). Patching the visibility back down prevents anyone else from using an AT on it and if we're making all fields public, then this field should not be non-final. I personally would prefer patching a second constructor into `TextureStateShard` which allows providing the runnables directly instead of having to override them afterwards. The patch could look something like this:
```patch
         public TextureStateShard(ResourceLocation pTexture, boolean pBlur, boolean pMipmap) {
-            super(() -> {
+            this(() -> {
                 TextureManager texturemanager = Minecraft.getInstance().getTextureManager();
                 texturemanager.getTexture(pTexture).setFilter(pBlur, pMipmap);
                 RenderSystem.setShaderTexture(0, pTexture);
             }, () -> {
-            });
+            }, pTexture, pBlur, pMipmap);
+        }
+        protected TextureStateShard(Runnable setupState, Runnable clearState, ResourceLocation pTexture, boolean pBlur, boolean pMipmap) {
+            super(setupState, clearState);
             this.texture = Optional.of(pTexture);
             this.blur = pBlur;
             this.mipmap = pMipmap;
         }
```

Reply:
While that would work in most cases, Neos `CustomizableTextureState` in `NeoForgeRenderTypes` requires the `setupState` to be non-final, as the modification made references `this`, which would not be possible via a `super(() -> { }, ...` since you cant reference `this` before `super`

https://github.com/neoforged/NeoForge/blob/c9c4d982b27128b390f6a6086426b2bf34cfc3e9/src/main/java/net/neoforged/neoforge/client/NeoForgeRenderTypes.java#L343-L353

notice the `this.` within the runnable, that would not be possible if passed via `super`.
`setupState` needs to be made non-final which due to the AT bug has to be done via a patch currently, unless I am mistaken that is?

Reply:
Dammit, yeah, you're right, it doesn't allow that. Another option that wouldn't even require a patch is to override `setupRenderState()` and perform the setup there, leaving the original setup runnable unused.